File: src/configuration.js
const fs = require("fs");

var data = {};

module.exports = {
    read: function() {
        var contents = "";
        try {
            contents = fs.readFileSync("config.json");
        }
        catch (e) {}

        if (contents.length > 0) {
            data = JSON.parse(contents);
        }
        else {
            data = {
                serverName: "Untitled RNet Controller",
                serverHost: false,
                serverPort: 3000,
                serialDevice: "/dev/tty-usbserial1"
            }
        }
    },
    write: function() {
        fs.writeFileSync("config.json", JSON.stringify(data));
    },
    get: function(key) {
        return data[key];
    }
}

File: src/RNetPi.Core/Models/Configuration.cs
namespace RNetPi.Core.Models;

public class Configuration
{
    public string ServerName { get; set; } = "Untitled RNet Controller";
    public string? ServerHost { get; set; }
    public int ServerPort { get; set; } = 3000;
    public string? WebHost { get; set; }
    public int? WebPort { get; set; }
    public string SerialDevice { get; set; } = "/dev/tty-usbserial1";
    public string WebHookPassword { get; set; } = string.Empty;
    public bool Simulate { get; set; } = false;
}